app/resources/v1/__tests__/user.test.js

```diff
@@ -1,68 +1,58 @@
 /* eslint-env jest */
 import request from 'supertest'
-import express from 'express'
+import { setupMockServer } from '../../../../test/helpers/setup-mock-server'
 import user from '../user'
 
 jest.mock('../../../models/user')
-jest.mock('../../../../lib/logger', () => function () {
-  return {
-    info: function () {},
-    error: function () {},
-    debug: function () {}
-  }
-})
-
-// Fake user info to test the API
-const userData = {
-  twitter: {
-    screenName: 'oluwaseun',
-    userId: '4232',
-    oauthAccessTokenKey: '461-Hvsiosdfsoafsoafedfd',
-    oauthAccessTokenSecret: 'WoZytO4kMLuafdafjdafja'
-  }
-}
-
-function setLoginToken (req, res, next) {
-  req.loginToken = '133e5110-5d2e-11e8-a8fd-678b57961690'
-  req.userId = 'oluwaseun'
-  next()
-}
+jest.mock('../../../../lib/logger')
 
-function setupMockServer () {
-  const app = express()
-
-  app.use(express.json())
+describe('PUT api/v1/users/:user_id', () => {
+  const app = setupMockServer((app) => {
+    app.put('/api/v1/users/:user_id', user.put)
+  })
 
-  app.get('/api/v1/users/:user_id', user.get)
-  // Set loginToken before running remaining endpoint
-  app.use(setLoginToken)
-  app.put('/api/v1/users/:user_id', user.put)
-  return app
-}
+  it('should respond with 204 user updates their own credentials', () => {
+    return request(app)
+      .put('/api/v1/users/user1')
+      .set('Authorization', 'Streetmix realm="" loginToken="xxxxxxxx-xxxx-xxxx-xxxx-1111111111111" userId="user1"')
+      .type('json')
+      .send(JSON.stringify({}))
+      .then((response) => {
+        expect(response.statusCode).toEqual(204)
+      })
+  })
 
-describe('PUT api/v1/users/:user_id', function () {
-  const app = setupMockServer()
+  it('should respond with 401 if a user PUTs to a different user', () => {
+    return request(app)
+      .put('/api/v1/users/user2')
+      .set('Authorization', 'Streetmix realm="" loginToken="xxxxxxxx-xxxx-xxxx-xxxx-2222222222222" userId="user1"')
```

Comment:
Would the `loginToken` end in 1s instead of 2s since it belongs to user1? It looks like it doesn't affect the test since finding a user by login token returns user1 by default, but to make it consistent we should change to 1s?